### PR TITLE
13856 upload component

### PR DESCRIPTION
**Pull request by unodgs**

### CODE REVIEW

File: src/common/api/common-resource-service.ts [hunk 1]
```diff
@@ -68,10 +68,10 @@ export class CommonResourceService<T extends Resource> {
         this.resourceType = '/' + resourceType + '/';
     }
 
-    create(data: Partial<T>) {
+    create(data?: Partial<T> | any) {
```
Comment (burzomir) [inline src/common/api/common-resource-service.ts]: TODO: refactor typings

File: src/common/api/filter-builder.ts [hunk 2]
```diff
@@ -3,38 +3,37 @@
 // SPDX-License-Identifier: AGPL-3.0
 
 import * as _ from "lodash";
-import { Resource } from "../../models/resource";
 
-export class FilterBuilder<T extends Resource = Resource> {
-    static create<T extends Resource = Resource>(resourcePrefix = "") {
-        return new FilterBuilder<T>(resourcePrefix);
+export class FilterBuilder {
```
Comment (burzomir) [inline src/common/api/filter-builder.ts]: TODO: Why not use a generic type?

File: src/store/collections/uploader/collection-uploader-reducer.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            startTime: 0,
+            prevTime: 0,
+            currentTime: 0
+        })),
+        START_UPLOAD: () => {
+            const startTime = Date.now();
+            return state.map(f => ({...f, startTime, prevTime: startTime}));
+        },
+        SET_UPLOAD_PROGRESS: ({ fileId, loaded, total, currentTime }) => {
+            const files = _.cloneDeep(state);
```
Comment (burzomir) [inline src/store/collections/uploader/collection-uploader-reducer.ts]: Why not to use array map the same way as in SET_UPLOAD_FILES? `files.map(file => file.id === fileId ? {...file, ...} : file)`

File: src/views-components/dialog-create/dialog-collection-create.tsx [hunk 4]
```diff
@@ -73,7 +84,7 @@ export const DialogCollectionCreate = compose(
                     maxWidth='sm'
                     disableBackdropClick={true}
                     disableEscapeKeyDown={true}>
-                    <form onSubmit={handleSubmit((data: any) => onSubmit(data))}>
+                    <form onSubmit={handleSubmit((data: any) => onSubmit(data, files))}>
```
Comment (burzomir) [inline src/views-components/dialog-create/dialog-collection-create.tsx]: Avoid creating functions in component props. In this case you could pass `onSubmit` to reduxForm config and call props.handleSubmit on button click.

File: src/views-components/update-collection-dialog/update-collection-dialog..tsx [hunk 5]
```diff
@@ -7,18 +7,18 @@ import { Dispatch } from "redux";
...
+    open: state.collections.updater.opened
 });
 
 const mapDispatchToProps = (dispatch: Dispatch) => ({
     handleClose: () => {
-        dispatch(collectionUpdatorActions.CLOSE_COLLECTION_UPDATOR());
+        dispatch(collectionUpdaterActions.CLOSE_COLLECTION_UPDATER());
     },
     onSubmit: (data: { name: string, description: string }) => {
         return dispatch<any>(editCollection(data))
```
Comment (burzomir) [inline src/views-components/update-collection-dialog/update-collection-dialog..tsx]: TODO: We should find out how to get rid of any type annotation when using dispatch with thunk action

Comment (unodgs) [reply src/views-components/update-collection-dialog/update-collection-dialog..tsx]: Yes, definitely

**Pull request merged**